### PR TITLE
Make MutationObserver polyfill aware of the shadow dom polyfill

### PR DESCRIPTION
**Pull request by arv**

### CODE REVIEW

File: MutationObserver.js [hunk 1]
```diff
@@ -343,6 +356,8 @@
     },
 
     addListeners_: function(node) {
+      node = wrap(node);
```
Comment (dfreedm) [inline MutationObserver.js:359]: This is failing assertions because the node is already wrapped.

Comment (arv) [inline MutationObserver.js:359]: That worked for me. Do you know how you got there?

Comment (dfreedm) [inline MutationObserver.js:359]: Slightly complicated. I forced toolkit/platform/ShadowDOM to your commit, then loaded Sandbox in IE 10 with the debug queryString. IE threw on that line from your assertion.

**Comment by arv** [comment]
Closing this in favor of #4

**Pull request closed without merging**